src/drive/mobile/lib/replication.js

```diff
@@ -1,22 +1,63 @@
-/* global cozy */
+/* global cozy emit */
 import { clientRevokedMsg } from './cozy-helper'
 
 export const startReplication = async (
+  hasIndexes,
   hasFinishedFirstReplication,
   firstReplicationFinished,
   refreshFolder,
-  revokeClient
+  revokeClient,
+  indexesCreated
 ) => {
   try {
+    let db, folderIndex
+    if (!hasIndexes) {
+      db = cozy.client.offline.getDatabase('io.cozy.files')
+      folderIndex = await db.createIndex({
+        index: { fields: ['dir_id', 'type', 'name'] }
+      })
+      const ddoc = {
+        _id: '_design/my_index',
+        views: {
+          recent_files: {
+            map: function(doc) {
+              if (!doc.trashed && doc.type !== 'directory') emit(doc.updated_at)
```

Comment:
What is this `emit` ?

Reply:
`emit` is used for map functions of map/reduce queries. This is a global that is made available by PouchDB to the map functions.

Reply:
TIL